diff --git a/src/modules/wireplumber.cpp b/src/modules/wireplumber.cpp
index a43ad29b..b30961cb 100644
--- a/src/modules/wireplumber.cpp
+++ b/src/modules/wireplumber.cpp
@@ -200,6 +200,7 @@ void waybar::modules::Wireplumber::onMixerChanged(waybar::modules::Wireplumber*
           return;
         }
       }
+      return;
     }

     spdlog::warn("[{}]: (onMixerChanged: {}) - Object with id {} not found", self->name_,
