diff --git a/notifier/libnotify.go b/notifier/libnotify.go
index 9fc43a1..ec47fbb 100644
--- a/notifier/libnotify.go
+++ b/notifier/libnotify.go
@@ -53,7 +53,7 @@ func setupLibnotifyNotifier(notifiers *sync.Map, title, body string) {

 	notification := notify.Notification{
 		AppName: "yubikey-touch-detector",
-		AppIcon: "yubikey-touch-detector",
+		AppIcon: "stock_lock",
 	}
 	notification.AddHint(notify.Hint{ID: "transient", Variant: dbus.MakeVariant(true)})
